### PR TITLE
Fix HistoryLocation in React v0.13 Example

### PR DESCRIPTION
**Pull request by ericclemmons**
Hi!

Using HistoryLocation, instead of RefreshLocation as used in the examples, I'm only able to navigate via react-router (both Link component and transitionTo('<page>')) once pr. page load. I've tried to fiddle around with both the source for react-router and react-resolver, but without success. The URL changes, but the 

I've reduced my issue down to [this repo](https://github.com/ostrgard/react-router-resolver-issue), which can be run with `npm install` and `npm run watch`. If you uncomment 11th line and comment the 10th line in `src/client.js`, you'll notice that routing works, but obviously serverside rendering does not. This leads me to think this could be a Resolver problem.

Anyone having the same issue? Have I done something wrong? I've tried to implement Resolver as per the examples as close as possible.

### CODE REVIEW

File: src/Resolver.js [hunk 1]
```diff
@@ -29,25 +30,23 @@ export default class Resolver extends React.Component {
     resolve: React.PropTypes.object,
   }
 
-  static render = function(render, node) {
-    const initialData = window.__REACT_RESOLVER_PAYLOAD__;
-
+  static render = function(render, node, data = window[PAYLOAD]) {
```
Comment (ericclemmons) [inline src/Resolver.js]: This was the crux of the solution: If the DOM has content but the payload is missing, _fetch the payload & call `Resolver.render` with it :)

Sometimes the best answer for recursion is recursion!

**Comment by ericclemmons** [comment]
This is nearly fixed, but I've also cleaned up the examples a bit as well.

**Comment by ericclemmons** [comment]
Confirmed working!  I'll merge & patch release this shortly...

**Comment by ericclemmons** [comment]
So sleepy...I'll do this tomorrow.

**Comment by ostrgard** [comment]
Happy you found a solution. Sleep tight!

**Comment by ericclemmons** [comment]
Uh...  shoot.  You shouldn't have closed this.

**Comment by ostrgard** [comment]
Please forgive my lack of Github skills.

**Pull request merged**